ansible_galaxy/models/collection_info.py

```diff
@@ -1,17 +1,79 @@
-
-import logging
+from __future__ import print_function
 
 import attr
+import json
+import logging
+import os
+import re
+import semver
 
-log = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
+TAG_REGEXP = re.compile('^[a-z0-9]+$')
 
 # see https://github.com/ansible/galaxy/issues/957
+
+
 @attr.s(frozen=True)
 class CollectionInfo(object):
-    namespace = attr.ib()
-    name = attr.ib()
-    version = attr.ib()
-    format_version = attr.ib(default=0.0)
-    author = attr.ib(default=None)
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+    authors = attr.ib(default=[])
     license = attr.ib(default=None)
+    description = attr.ib(default=None)
+    keywords = attr.ib(default=[])
+    readme = attr.ib(default='README.md')
+    dependencies = attr.ib(default=[])
+
+    @property
+    def namespace(self):
+        return self.name.split('.', 1)[0]
+
+    @name.validator
+    @version.validator
+    @license.validator
+    @description.validator
+    def _check_required(self, attribute, value):
```

Comment:
What is the custom required validator for?  Not setting a default should make a field required by default iirc.

Reply:
Without a validator, we get an error message that provides 0 context, and leaves the user helpless to solve the problem. The point here is to provide better UX.